src/pages/namespace/Settings/index.tsx

```diff
@@ -1,3 +1,4 @@
+import Broadcasts from "./Broadcasts";
 import { FC } from "react";
```

Comment:
I just had an idea, and I would love to hear your feedback on this (because I'm not sure about it yet). The settings page has a lot of components and each of them load data and will populate its content when all data was fetched. Each of this data population is very likely to happen in its own render loop. So there is a lot of flickering on the initial page load.

What do you think about the approach to wait for all data to be fetched before rendering any component? From the user perspective, it just removes the flickering, but I'm unsure about the DX/maintainability of this code:

```typescript
import Broadcasts from "./Broadcasts";
import { FC } from "react";
import RegistriesList from "./Registries";
import SecretsList from "./Secrets";
import VariablesList from "./Variables";
import { useBroadcasts } from "~/api/broadcasts/query/useBroadcasts";
import { useRegistries } from "~/api/registries/query/get";
import { useSecrets } from "~/api/secrets/query/get";
import { useVars } from "~/api/variables/query/useVariables";

// this hook uses all initial data fetching hooks from the components on
// this page and returns true if all of them are fetched. This is used to
// render the page only when all data is fetched and avoid flickering as
// all components are rendered at the same time and data comes in at different
// times.
const usePrefetchData = () => {
  const { isFetched: isFetchedisFetchedBroadcasts } = useBroadcasts();
  const { isFetched: isFetchedSecrets } = useSecrets();
  const { isFetched: isFetchedRegistries } = useRegistries();
  const { isFetched: isFetchedVars } = useVars();
  return (
    isFetchedisFetchedBroadcasts &&
    isFetchedSecrets &&
    isFetchedRegistries &&
    isFetchedVars
  );
};

const SettingsPage: FC = () => {
  const prefetched = usePrefetchData();

  if (!prefetched) {
    return null;
  }

  return (
    <div className="flex flex-col space-y-6 p-10">
      <section>
        <SecretsList />
      </section>

      <section>
        <RegistriesList />
      </section>

      <section>
        <VariablesList />
      </section>

      <section data-testid="broadcasts-section">
        <Broadcasts />
      </section>
    </div>
  );
};

export default SettingsPage;
``` 

**PRO**
- It is independent of the components and just prefills the cache, each of the components would still work perfectly fine with or without this code.

**CON:**
- this hook needs to be maintained, it's easy to forget about it when updating any component on that page (even if the possible "damage" will not break anything, but could make one or two request too much or too little)

Reply:
I think this is a clever idea and an elegant solution. I think the maintainability drawback is very minor because the hook is very easy to understand.

However, I was a bit worried that the page could be empty for too long if we wait until all data is fetched, so I played around with bandwidth throttling in devtools.

But even on slow 3G, all data is rendered almost instantly (it's just a few Kb that need to be fetched). But that also means there isn't a lot of noticeable flickering because everything loads so quickly. (The initial loading of the assets, however, takes very long, but that is a different story.)

In conclusion, I have no strong objections, but I am a bit hesitant about this because it adds more complexity to solve a problem that may not really exist.

Reply:
Thanks for that feedback. Personally, the flickering bothers me enough to actually vote for that caching solution.

Reply:
Ok, implemented in 24b00dbb465acbdcc314e9b82c82736bb6079e33.